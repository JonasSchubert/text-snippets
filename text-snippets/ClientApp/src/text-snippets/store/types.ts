export const ActionTypes = {
  add: 'add',
  delete: 'delete',
  load: 'load',
  update: 'update'
};

export const GetterTypes = {
  error: 'error',
  isLoading: 'isLoading',
  list: 'list'
};

export const ModuleType = 'text-snippets';

export const MutationTypes = {
  setError: 'setError',
  setIsLoading: 'setIsLoading',
  setList: 'setList'
};

export const RouteTypes = {
  textSnippets: 'TextSnippets'
};
