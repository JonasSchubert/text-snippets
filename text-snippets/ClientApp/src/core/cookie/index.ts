export const appName = "dotnetcore-vue-template";

export const daysTilExpired: { [key: string]: number } = {};

export const keys: { [key: string]: string } = {};
