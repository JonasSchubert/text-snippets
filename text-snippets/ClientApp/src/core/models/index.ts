export * from './root-state.d';
