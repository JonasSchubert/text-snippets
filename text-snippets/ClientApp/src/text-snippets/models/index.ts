export * from './text-snippet.d';
export * from './text-snippets-state.d';
