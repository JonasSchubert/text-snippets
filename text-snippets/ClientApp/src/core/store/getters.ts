import { GetterTree } from 'vuex';
import { RootState } from '@/core/models';
import { GetterTypes } from './types';

export const createGetters = (): GetterTree<RootState, RootState> => ({
  [GetterTypes.isAnythingLoading]: (state: RootState): boolean => state.requests > 0
});
