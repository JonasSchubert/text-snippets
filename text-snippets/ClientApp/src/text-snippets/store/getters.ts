import { GetterTree } from 'vuex';
import { RootState } from '@/core/models';
import { TextSnippet, TextSnippetsState } from '@/text-snippets/models';
import { GetterTypes } from './types';

export const createGetters = (): GetterTree<TextSnippetsState, RootState> => ({
  [GetterTypes.error]: (state: TextSnippetsState): Error | undefined => state.error,
  [GetterTypes.isLoading]: (state: TextSnippetsState): boolean => state.isLoading,
  [GetterTypes.list]: (state: TextSnippetsState): TextSnippet[] => state.list
});
