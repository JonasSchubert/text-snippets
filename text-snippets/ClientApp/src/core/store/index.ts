import Vue from 'vue';
import Vuex, { Store, StoreOptions } from 'vuex';
import { createI18nModule } from '@/core/i18n/module-helper';
import { RootState } from '@/core/models';
import { ModuleType as LocalStorageModule } from "@/libs/local-storage/store/types";
import { createLocalStorageModule } from "@/libs/local-storage/store";
import { createModule as createTextSnippetsModule } from '@/text-snippets/store';
import { ModuleType as TextSnippetsModule } from '@/text-snippets/store/types';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

Vue.use(Vuex);

export const createStore = (vuetify: any): Store<RootState> => {
  const store: StoreOptions<RootState> = {
    getters: createGetters(),
    modules: {
      i18n: createI18nModule(),
      [LocalStorageModule]: createLocalStorageModule<RootState>('text-snippets', vuetify),
      [TextSnippetsModule]: createTextSnippetsModule()
    },
    mutations: createMutations(),
    state: createState(),
    strict: process.env.NODE_ENV !== 'production'
  };

  return new Store<RootState>(store);
};
