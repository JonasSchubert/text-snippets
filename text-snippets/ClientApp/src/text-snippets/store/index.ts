import { Module } from 'vuex';
import { RootState } from '@/core/models';
import { TextSnippetsState } from '@/text-snippets/models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

export const createModule = (): Module<TextSnippetsState, RootState> => ({
  namespaced: true,
  actions: createActions(),
  getters: createGetters(),
  mutations: createMutations(),
  state: createState()
});
