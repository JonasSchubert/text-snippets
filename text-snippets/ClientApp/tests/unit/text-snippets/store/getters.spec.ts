import { RootState } from '@/core/models';
import { TextSnippet, TextSnippetsState } from '@/text-snippets/models';
import { createGetters } from '@/text-snippets/store/getters';
import { GetterTypes } from '@/text-snippets/store/types';

describe('text-snippets/getters', () => {
  let state: TextSnippetsState;
  const getters = createGetters();

  beforeEach(() => {
    state = {
      error: {
        message: 'message',
        name: 'name'
      },
      isLoading: true,
      list: [
        { id: '1' } as TextSnippet
      ]
    };
  });

  describe(`${GetterTypes.error}`, () => {
    test('should return error as expected', () => {
      // Arrange && Act
      const actual = getters[GetterTypes.error](state, getters, {} as RootState, {});

      // Assert
      expect(actual).toMatchSnapshot({
        message: 'message',
        name: 'name'
      });
    });
  });

  describe(`${GetterTypes.isLoading}`, () => {
    test('should return isLoading as expected', () => {
      // Arrange && Act
      const actual = getters[GetterTypes.isLoading](state, getters, {} as RootState, {});

      // Assert
      expect(actual).toBeTruthy();
    });
  });

  describe(`${GetterTypes.list}`, () => {
    test('should return list as expected', () => {
      // Arrange && Act
      const actual = getters[GetterTypes.list](state, getters, {} as RootState, {});

      // Assert
      expect(actual).toMatchSnapshot([
        { id: '1' } as TextSnippet
      ]);
    });
  });
});
