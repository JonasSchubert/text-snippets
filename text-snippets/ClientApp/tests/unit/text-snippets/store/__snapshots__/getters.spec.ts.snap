// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-snippets/getters error should return error as expected 1`] = `
Object {
  "message": "message",
  "name": "name",
}
`;

exports[`text-snippets/getters list should return list as expected 1`] = `
Object {
  "0": Object {
    "id": "1",
  },
}
`;
